### PR TITLE
apis for add product rate and product rates index

### PR DESCRIPTION
**Pull request by kandelchy**

### CODE REVIEW

File: app/controllers/api/v1/rates_controller.rb [hunk 1]
```diff
@@ -0,0 +1,40 @@
+class Api::V1::RatesController < ApplicationController
+
+  def index
+    if params[:ad_id].present?
```
Comment (ssharba) [inline app/controllers/api/v1/rates_controller.rb]: I would add begin rescue here

File: app/controllers/api/v1/rates_controller.rb [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+      @ad = Ad.find(params[:ad_id])
```
Comment (MwMaki) [inline app/controllers/api/v1/rates_controller.rb]: No need to use (@) sign anywhere

File: app/controllers/api/v1/rates_controller.rb [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+          codable_id = params[:rate][:product_id]
+        elsif params[:rate][:ad_id].present?
+          codable_type = 'Ad'
+          codable_id = params[:rate][:ad_id]
+        elsif params[:rate][:business_id].present?
+          codable_type = 'BusinessInfo'
+          codable_id = params[:rate][:business_id]
+        else
+          raise 'Rate codable type incorrect'
+          return
```
Comment (ssharba) [inline app/controllers/api/v1/rates_controller.rb]: no need for return as raise will brake the function will jump to rescue

File: app/models/rate.rb [hunk 4]
```diff
@@ -1,3 +1,4 @@
 class Rate < ApplicationRecord
   belongs_to :codable, polymorphic: true
+  has_one :report_abuse, dependent: :destroy
```
Comment (MwMaki) [inline app/models/rate.rb]: as long as no one allow to will delete the rate then I don't think we need the 
, dependent: :destroy

File: app/serializers/rate_serializer.rb [hunk 5]
```diff
@@ -0,0 +1,9 @@
+class RateSerializer < ActiveModel::Serializer
+  attributes :id, :customer_id, :rating, :codable_type, :codable_id, :review, :product_name
+  has_one :report_abuse
+
+  def product_name
+    Product.find(object.codable_id).name
```
Comment (MwMaki) [inline app/serializers/rate_serializer.rb]: Can't we use the association ?

File: app/controllers/api/v1/rates_controller.rb [hunk 6]
```diff
@@ -0,0 +1,41 @@
+class Api::V1::RatesController < ApplicationController
+
+  def index
+    begin
+      if params[:ad_id].present?
```
Comment (ssharba) [inline app/controllers/api/v1/rates_controller.rb]: @kandelchy don't we need pagination here?

**Pull request merged**